Assets/Scripts/Strategy/BaseManagement/BuildingFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BuildingFactory
+{
+    private static BuildingFactory instance;
+
+    private BuildingFactory() { }
+
+    public static BuildingFactory Instance
+    {
+        get
+        {
+            if (instance == null)
+            {
+                instance = new BuildingFactory();
+            }
+
+            return instance;
+        }
+    }
+
+    internal IBuilding CreateBarracks()
+    {
+        return new Barracks();
+    }
+
+    internal IBuilding CreateGranary()
+    {
+        return null;
+    }
+}
```

Comment:
Instance doesn't need to be publicly accessible and just makes the class harder to use. Make all functions static and have them call the private static instance. As of now though I don't know why this needs to be a singleton, it can just be a static class

Reply:
0e267b2